enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/slotted/helpers/SlottedPipeBuilderUtils.scala

```diff
@@ -201,4 +201,94 @@ object SlottedPipeBuilderUtils {
       case _ =>
         throw new InternalException(s"Do not know how to make setter for slot $slot")
     }
+
+  /**
+    * Use this to make a specialized setter function for a slot,
+    * that takes as input an ExecutionContext and a primitive long value.
+    */
+  def makeSetPrimitiveInSlotFunctionFor(slot: Slot, valueType: CypherType): (ExecutionContext, Long) => Unit =
+    (slot, valueType) match {
+      case (LongSlot(offset, nullable, CTNode), CTNode) =>
+        if (AssertionUtils.assertionsEnabled && !nullable) {
+          (context: ExecutionContext, value: Long) =>
+            if (value == -1L)
+              throw new ParameterWrongTypeException(s"Cannot assign null to a non-nullable slot")
+            context.setLongAt(offset, value)
+        }
+        else {
+          (context: ExecutionContext, value: Long) =>
+            context.setLongAt(offset, value)
+        }
+
+      case (LongSlot(offset, nullable, CTRelationship), CTRelationship) =>
+        if (AssertionUtils.assertionsEnabled && !nullable) {
+          (context: ExecutionContext, value: Long) =>
+            if (value == -1L)
+              throw new ParameterWrongTypeException(s"Cannot assign null to a non-nullable slot")
+            context.setLongAt(offset, value)
+        }
+        else {
+          (context: ExecutionContext, value: Long) =>
+            context.setLongAt(offset, value)
+        }
```

Comment:
I like the assertion controlled null-check! The new code here feel very copy-pasty though. Could we fold it somehow to avoid so much duplication?